crates/ty_python_semantic/src/types/display.rs

```diff
@@ -59,6 +65,89 @@ impl fmt::Debug for DisplayType<'_> {
     }
 }
 
+#[derive(Copy, Clone)]
+pub struct QualifiedDisplayType<'db> {
+    ty: &'db Type<'db>,
+    db: &'db dyn Db,
+}
+
+impl Display for QualifiedDisplayType<'_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self.ty {
+            Type::ClassLiteral(literal) => {
+                write!(f, "{}", self.get_qualified_representation(*literal))
```

Comment:
this can be slightly more performant than using the macro (though I'd still use the macro whenever it makes the code cleaner, e.g. in your `EnumLiteral` branch below)

```suggestion
                f.write_str(&self.get_qualified_representation(*literal))
```